### PR TITLE
core(config): faster category validation

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
On my machine 90% of the non-requireing time spent in config:init was just validating the categories. Simple conversion of this `.find` to a map eliminated ~60ms.

**Related Issues/PRs**
#6442

### CODE REVIEW

File: lighthouse-core/config/config.js [hunk 1]
```diff
@@ -64,13 +64,18 @@ function validateCategories(categories, audits, groups) {
     return;
   }
 
+  const auditsKeyedById = new Map((audits || []).map(audit =>
+    /** @type {[string, LH.Config.AuditDefn]} */
+    ([audit.implementation.meta.id, audit]))
+  );
```
Comment (Hoten) [inline lighthouse-core/config/config.js]: the parens here are confusing - it led me to think `Map` had a `.map` function.

is there an alternative way to balance these parentheses?

Comment (patrickhulce) [reply lighthouse-core/config/config.js]: Oh haha, whoops yeah I'll move the second one down :)

**Comment by paulirish** [comment]
60ms! Hot damn.

Nice one.

**Comment by patrickhulce** [comment]
> 60ms! Hot damn.

I can't tell if that's sarcasm or not 😆

**Comment by paulirish** [comment]
>> 60ms! Hot damn.

> I can't tell if that's sarcasm or not 😆

It's not. I think this points to the wins of profiling-guided development. Pretty awesome.

**Pull request merged**